app/src/pencil2d.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+
+Pencil2D - Traditional Animation Software
+Copyright (C) 2005-2007 Patrick Corrieri & Pascal Naidon
+Copyright (C) 2012-2020 Matthew Chiawen Chang
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+*/
+
+#include <memory>
+
+#include <QDebug>
+#include <QFileOpenEvent>
+#include <QIcon>
+#include <QLibraryInfo>
+#include <QSettings>
+#include <QTranslator>
+
+#include "commandlineexporter.h"
+#include "commandlineparser.h"
+#include "mainwindow2.h"
+#include "pencildef.h"
+#include "platformhandler.h"
+
+#include "pencil2d.h"
+
+Pencil2D::Pencil2D(int& argc, char** argv) :
+    QApplication(argc, argv)
+{
+    // Set organization and application name
+    setOrganizationName("Pencil2D");
+    setOrganizationDomain("pencil2d.org");
+    setApplicationName("Pencil2D");
+    setApplicationDisplayName("Pencil2D");
+
+    // Set application version
+    setApplicationVersion(APP_VERSION);
+
+    // Set application icon
+    setWindowIcon(QIcon(":/icons/icon.png"));
+
+    // Associate the application with our desktop entry
+    setDesktopFileName("org.pencil2d.Pencil2D.desktop");
+
+    installTranslators();
+}
+
+Status Pencil2D::handleCommandLineOptions()
+{
+    CommandLineParser parser;
+    parser.process(arguments());
+
+    QString inputPath = parser.inputPath();
+    QStringList outputPaths = parser.outputPaths();
+
+    if (outputPaths.isEmpty())
+    {
+        prepareGuiStartup(inputPath);
+        return Status::OK;
+    }
+
+    auto mainWindow = new MainWindow2;
```

Comment:
Having `std::unique_ptr` here? probably making no difference in the context, just a good style though.